File: pset6/readability.py
text = input("Text: ")

letters = 0
words = 1
sentences = 0

letters_string = ""
space_string = ""
punct_string = ""
special_string = ""

for i in range(len(text)):
    if text[i].isalpha() == True:
        letters_string += text[i]
        letters = letters + 1

    elif text[i] == " ":
        space_string += "/"
        words = words + 1

    elif text[i] == "!" or text[i] == "?" or text[i] == ".":
        punct_string += text[i]
        sentences = sentences + 1

    else:
        special_string += text[i]
        continue


l = float(letters) / words * 100
s = float(sentences) / words * 100

result = 0.0588 * float(l) - 0.296 * float(s) - 15.8
result = round(result)

if result > 16:
    print("Grade 16+")

elif result < 1:
    print("Before Grade 1")

else:
    print(f"Grade {result}")

File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    //Get user input
    string text = get_string("Text: ");
    unsigned int len = strlen(text);

    float letter_count = 0;
    float word_count = 1;
    float sentence_count = 0;

    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            letter_count++;
        }

        else if (text[i] == ' ')
        {
            word_count++;
        }

        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count++;
        }

        else
        {
            continue;
        }
    }

    float ratio = 100 / word_count;
    float S = sentence_count * ratio;
    float L = letter_count * ratio;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else if (index < 0)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}